raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -1990,7 +1995,7 @@ private long pollLeader(long currentTimeMs) {
         LeaderState<T> state = quorum.leaderStateOrThrow();
         maybeFireLeaderChange(state);
 
-        if (shutdown.get() != null || state.isResignRequested()) {
+        if (shutdown.get() != null || state.isResignRequested() || state.hasMajorityFollowerFetchExpired(currentTimeMs)) {
```

Comment:
In line [2014](https://github.com/apache/kafka/pull/14428/files#diff-1da15c51e641ea46ea5c86201ab8f21cfee9e7c575102a39c7bae0d5ffd7de39R2014) KRaft calculates the next time it should poll the leader.
```java
        return Math.min(timeUntilFlush, timeUntilSend);
```

If you extend this to something like below, it should wake up the leader when the timer has expired.
```java
        return Math.min(timeUntilFlush, timeUntilSend, state.checkQuorumExpiry());
```

Reply:
Ah, good catch! Updated the `hasMajorityFollowerFetchExpired` to `timeUntilCheckQuorumExpires` now.